### PR TITLE
Validate another layer of fields and some cleanup\rTESTING=unit

### PR DESCRIPTION
**Pull request by scottmcmaster**
This is enough validation to ensure that (for example) content ids are strings. I also refactored the validator to be a class so that we can easily replace it dynamically (for example, with a faster or no-op validator), consolidated with the badly-named "Settings" class, and removed a bogus dependency on byebug.

### CODE REVIEW

File: lib/promoted/ruby/client/validator.rb [hunk 1]
```diff
@@ -2,26 +2,151 @@ module Promoted
...
+                end
+            end
+
+            def validate_request!(req)
+                validate_fields!(
+                    req,
+                    "request",
+                    [
+                        {
+                            :name => :platform_id,
```
Comment (danbosnichill) [inline lib/promoted/ruby/client/validator.rb:85]: side-comment - Right now, we're not forcing clients to set this.  We're just doing it on the server-side.  It's fine to leave it in.  We'll eventually ask for it.

File: lib/promoted/ruby/client/validator.rb [hunk 2]
```diff
@@ -2,26 +2,151 @@ module Promoted
...
+                validate_fields!(
+                    ins,
+                    "insertion",
+                    [
+                        {
+                            :name => :platform_id,
+                            :type => Integer
+                        },
+                        {
+                            :name => :insertion_id,
```
Comment (danbosnichill) [inline lib/promoted/ruby/client/validator.rb:44]: side-note - clients of the library should not set request_id, insertion_id or delivery_score.

**Review by danbosnichill — APPROVED** [review]

File: lib/promoted/ruby/client/validator.rb (hunk 1 above)
Comment (scottmcmaster) [reply lib/promoted/ruby/client/validator.rb:85]: Ack. I was mostly trying to mimic the TypeScript annotations' behaviors with the field validators.

File: lib/promoted/ruby/client/validator.rb (hunk 2 above)
Comment (scottmcmaster) [reply lib/promoted/ruby/client/validator.rb:44]: Ah, I think in that case a check for delivery_score needs to be added to check_that_log_ids_not_set!, which I did. Thanks.

**Pull request merged**